src/parser/grammar.ne

```diff
@@ -23,11 +23,20 @@ const toKeywordNode = (token: Token): KeywordNode => ({
   raw: token.raw,
 });
 
-const addLeadingComments = (node: AstNode, comments: CommentNode[]): AstNode =>
-  comments.length > 0 ? { ...node, leadingComments: comments } : node;
-
-const addTrailingComments = (node: AstNode, comments: CommentNode[]): AstNode =>
-  comments.length > 0 ? { ...node, trailingComments: comments } : node;
+interface CommentAttachments {
+  leading?: CommentNode[];
+  trailing?: CommentNode[];
+}
+
+const addComments = (node: AstNode, { leading, trailing = [] }: CommentAttachments): AstNode => {
```

Comment:
Should both or neither of `leading` and `trailing` have empty array initialisers?

Reply:
Oops... this default value was indeed forgotten in.